src/Symfony/Component/Security/Http/Attribute/IsGrantedContext.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Attribute;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecision;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+readonly class IsGrantedContext implements AuthorizationCheckerInterface
```

Comment:
This class should either be final or should not be `readonly`, because of the impact the `readonly` keyword has on inheritance (there is *no* way to add or remove the `readonly` keyword on a non-final class in a BC way, so this would put a constraint on Symfony until version +Inf)

Reply:
Just asking: is it OK to add (some) logic to a context class (such as the `isGranted()` method here)? I thought these context objects are more similar to DTOs or value objects and just contain data and some lightweight methods to access to that.

Reply:
This is not strictly speaker a DTO or a VO. We do what we want in the end. Here this provides helpers, exactly like expression-language providers.

Reply:
I tend to agree but here the method is just one line with 0 logic, so it feels OK to me in this case

Reply:
`AuthorizationChecker` already hides `AccessDecisionManager` behind a simple API that people already know about so let me ask, do we really need that new broadly-scoped context object people will have to learn about?

Reply:
What's the alternative you're suggesting?
On my side: when using EL, one can use nice helpers and variables.
When using a closure, the DX is worse at the moment. Passing many args is a mess because order, optional ones, unused ones. Not to say helpers which cannot be provided.
This simple context object solves all those issues.